File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speeds/aac/AACYPort2.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speeds.aac;

import net.ccbluex.liquidbounce.event.MoveEvent;
import net.ccbluex.liquidbounce.features.module.modules.movement.speeds.SpeedMode;
import net.ccbluex.liquidbounce.utils.MovementUtils;

public class AACYPort2 extends SpeedMode {

    public AACYPort2() {
        super("AACYPort2");
    }

    @Override
    public void onMotion() {
        if(MovementUtils.isMoving()) {
            mc.thePlayer.cameraPitch = 0F;

            if(mc.thePlayer.onGround) {
                mc.thePlayer.jump();
                mc.thePlayer.motionY = 0.3851F;
                mc.thePlayer.motionX *= 1.01;
                mc.thePlayer.motionZ *= 1.01;
            }else
                mc.thePlayer.motionY = -0.21D;
        }
    }

    @Override
    public void onUpdate() {
    }

    @Override
    public void onMove(MoveEvent event) {
    }
}

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speeds/aac/AACYPort2.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speeds.aac

import net.ccbluex.liquidbounce.event.MoveEvent
import net.ccbluex.liquidbounce.features.module.modules.movement.speeds.SpeedMode
import net.ccbluex.liquidbounce.utils.MovementUtils

class AACYPort2 : SpeedMode("AACYPort2") {
    override fun onMotion() {
        if (MovementUtils.isMoving) {
            val thePlayer = mc.thePlayer ?: return

            thePlayer.cameraPitch = 0f
            if (thePlayer.onGround) {
                thePlayer.jump()
                thePlayer.motionY = 0.3851
                thePlayer.motionX *= 1.01
                thePlayer.motionZ *= 1.01
            } else thePlayer.motionY = -0.21
        }
    }

    override fun onUpdate() {}
    override fun onMove(event: MoveEvent) {}
}